### DISCUSSION TITLE
Request for Documentation on View Queries

### DISCUSSION THREAD

**Discussion by hugo-abdou** [pocketbase/pocketbase#5079] [Q&A] [answered]
Hello guys,
I hope you are doing well. I have been working on your project and appreciate the value it brings. However, I have encountered some challenges related to view queries.

Despite my efforts, I have been unable to find a section in the documentation that covers how to create and execute view queries. The queries I have attempted to run so far have not been successful.

Could you please direct me to any resources, documentation, or examples that provide guidance on crafting efficient and effective view queries? Any assistance or pointers would be greatly appreciated.

Thank you for your time and support.

Best regards,
Hugo

**Comment by ganigeorgiev** [comment]
> The queries I have attempted to run so far have not been successful.

Please provide concrete examples. All SQLite `SELECT` statements are supported with the following caveats:

- Top-level `UNION` expressions are not supported. A workaround is to wrap the query in a subquery (eg. `SELECT id, someField FROM (...)`).
- Wildcard (`*`) columns are not supported.
- The query must always have a unique `id` column.
- Each selected column must be a proper PocketBase collection field name.

**Reply by ganigeorgiev** [reply, marked as answer]
To confirm that your query is actually a valid SQLite statement and not a PocketBase issue, you can execute it using the sqlite3 cli tool directly against the `pb_data/data.db` file:

```sh
sqlite3 pb_data/data.db 'your query'
```

**Reply by hugo-abdou** [reply]
thank you @ganigeorgiev I'm trying to switch my business to pocketbase i like it a lot but sometimes I get stuck,
it's good to have someone to help from a time to another,
thank you again sir
